test/IPCSocket.h

```diff
@@ -0,0 +1,118 @@
+/*
+	This file is part of cpp-ethereum.
+
+	cpp-ethereum is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	cpp-ethereum is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/** @file IPCSocket.h
+ * @author Dimtiry Khokhlov <dimitry@ethdev.com>
+ * @date 2016
+ */
+
+#include <string>
+#include <stdio.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <boost/test/unit_test.hpp>
+#include <libdevcore/CommonIO.h>
+
+using namespace std;
+
+class IPCSocket
+{
+public:
+	IPCSocket(string const& _address) : m_address(_address)
+	{
+		m_saun.sun_family = AF_UNIX;
+		strcpy(m_saun.sun_path, _address.c_str());
+
+		if ((m_socket = socket(AF_UNIX, SOCK_STREAM, 0)) < 0)
+			BOOST_FAIL("Error creating IPC socket object");
+
+		int len = sizeof(m_saun.sun_family) + strlen(m_saun.sun_path);
+
+		if (connect(m_socket, (const struct sockaddr *)&m_saun, len) < 0)
+			BOOST_FAIL("Error connecting to IPC socket: " << _address);
+
+		m_fp = fdopen(m_socket, "r");
+	}
+
+	string sendRequest(string const& _req)
+	{
+		send(m_socket, _req.c_str(), _req.length(), 0);
+
+		char c;
+		string response;
+		while ((c = fgetc(m_fp)) != EOF)
+		{
+			if (c != '\n')
+				response += c;
+			else
+				break;
+		}
+		return response;
+	}
+
+	~IPCSocket() { close(m_socket); }
+
+private:
+	FILE *m_fp;
+	string m_address;
+	struct sockaddr_un m_saun;
+	int m_socket;
+
+};
+
+class RPCRequest
```

Comment:
I am planning to move anything related to construction of RPC requests to RPCRequest or whatever name would be appropriate